src/components/forms/DomainSettings/SourceCodeIdRow.tsx

```diff
@@ -0,0 +1,78 @@
+import { ANT_LUA_ID } from '@ar.io/sdk';
+import { Tooltip } from '@src/components/data-display';
+import ArweaveID, {
+  ArweaveIdTypes,
+} from '@src/components/layout/ArweaveID/ArweaveID';
+import UpgradeAntModal from '@src/components/modals/ant-management/UpgradeAntModal/UpgradeAntModal';
+import { useANTLuaSourceCode } from '@src/hooks/useANTLuaSourceCode';
+import { ArweaveTransactionID } from '@src/services/arweave/ArweaveTransactionID';
+import { useState } from 'react';
+
+import DomainSettingsRow from './DomainSettingsRow';
+
+export default function SourceCodeIdRow({
+  sourceCodeTxId,
+  editable,
+  antId,
+}: {
+  antId?: string;
+  sourceCodeTxId?: string;
+  editable: boolean;
+}) {
+  const { data } = useANTLuaSourceCode();
+  const [showUpgradeAntModal, setShowUpgradeAntModal] = useState(false);
+  return (
+    <>
+      <DomainSettingsRow
+        label="Source Code TX ID:"
+        editable={true}
+        value={
+          <span
+            className="flex center"
+            style={{
+              justifyContent: 'flex-start',
+              gap: '10px',
+            }}
+          >
+            {sourceCodeTxId ? (
+              <ArweaveID
+                id={new ArweaveTransactionID(sourceCodeTxId)}
+                shouldLink
+                type={ArweaveIdTypes.TRANSACTION}
+              />
+            ) : (
+              'N/A'
+            )}
+          </span>
+        }
+        action={
+          editable &&
+          sourceCodeTxId &&
+          ![ANT_LUA_ID, data?.originalTxId]
+            .filter((id) => id !== undefined)
+            .includes(sourceCodeTxId) && [
```

Comment:
It's not implicit what this means. Can you add this to a variable with a meaningful name in this context?